### PR TITLE
CXX-547 de-inline json.hpp

### PR DESCRIPTION
**Pull request by amidvidy**
need this for CXX-489.

### CODE REVIEW

**Comment by acmorrow** [comment]
@amidvidy something looks wrong about this PR.

**Comment by amidvidy** [comment]
fixed pr and added MONGOCXX_API.

File: src/bsoncxx/json.cpp [hunk 1]
```diff
@@ -0,0 +1,267 @@
+// Copyright 2014 MongoDB Inc.
```
Comment (acmorrow) [inline src/bsoncxx/json.cpp]: 2015 now

File: src/bsoncxx/json.cpp [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+            pad();
+            out << "]";
+        }
+
+        void visit_value(const types::b_binary& value) {
+
+            std::size_t b64_len;
+
+            b64_len = (value.size / 3 + 1) * 4 + 1;
+            std::unique_ptr<char> b64(reinterpret_cast<char*>(operator new(b64_len)));
```
Comment (acmorrow) [inline src/bsoncxx/json.cpp]: Is there any reason that this isn't:

auto b64 = stdx::make_unique<char>(b64_len);

File: src/bsoncxx/json.cpp [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+            visit_children(value.value);
+            pad();
+            out << "]";
+        }
+
+        void visit_value(const types::b_binary& value) {
+
+            std::size_t b64_len;
+
+            b64_len = (value.size / 3 + 1) * 4 + 1;
```
Comment (acmorrow) [inline src/bsoncxx/json.cpp]: Why?

File: src/bsoncxx/json.cpp (hunk 2 above)
Comment (amidvidy) [inline src/bsoncxx/json.cpp]: I'm going to defer to @hanumantmk, he wrote this code, I'm just shuffling it around.

File: src/bsoncxx/json.cpp (hunk 3 above)
Comment (amidvidy) [inline src/bsoncxx/json.cpp]: @hanumantmk?

File: src/bsoncxx/json.cpp (hunk 2 above)
Comment (amidvidy) [inline src/bsoncxx/json.cpp]: i think that would make sense though.

Comment (hanumantmk) [inline src/bsoncxx/json.cpp]: We can certainly change it over.  I think I had a bit of a psychological hangup over using new[] for things I associate with malloc when I was doing bsoncxx.

There's no real rational reason to use operator new() here though, definitely change it

File: src/bsoncxx/json.cpp (hunk 3 above)
Comment (hanumantmk) [inline src/bsoncxx/json.cpp]: the size computation?

It's how large things can get after you run them through ntop.  You started with binary and you're going to a string encoding, so you get a growth rate associated with going from 8bits/byte to 6bits/byte.  I think I cribbed that particular arithmetic from somewhere reputable, but I can't quite remember where now.

**Comment by acmorrow** [comment]
LGTM

**Pull request merged**